pymongo/periodic_executor.py

```diff
@@ -23,9 +23,102 @@
 import weakref
 from typing import Any, Optional
 
-from pymongo.lock import _create_lock
+from pymongo.lock import _ALock, _create_lock
 
-_IS_SYNC = True
+_IS_SYNC = False
+
+
+class AsyncPeriodicExecutor:
+    def __init__(
+        self,
+        interval: float,
+        min_interval: float,
+        target: Any,
+        name: Optional[str] = None,
+    ):
+        """Run a target function periodically on a background thread.
```

Comment:
The docs in this method still mention threads.  There's also an extraneous comment block in this method.